sshd-core/src/test/java/org/apache/sshd/server/auth/AsyncAuthTestBase.java

```diff
@@ -0,0 +1,142 @@
+    public AsyncAuthTestBase() {
+    }
+
+    public void startServer() throws Exception {
+        startServer(null);
+    }
+
+    public void startServer(Integer timeout) throws Exception {
+        if (server != null) {
+            fail("Server already started");
+        }
+        server = SshServer.setUpDefaultServer();
+        if (timeout != null) {
+            server.getProperties().put(FactoryManager.AUTH_TIMEOUT, timeout.toString());
+        }
+        server.setPort(5000);
+        server.setKeyPairProvider(new SimpleGeneratorHostKeyProvider(new File("hostkey.ser").toPath()));
+        server.setPasswordAuthenticator((username, password, session) -> authenticator.authenticate(username, password, session));
+        server.setShellFactory(new EchoShellFactory());
+        server.start();
+    }
+
+    @After
+    public void stopServer() throws Exception {
+        if (server != null) {
+            server.stop();
+        }
+        server = null;
+    }
+
+    @Test
+    public void testSyncAuthFailed() throws Exception {
+        startServer();
+        authenticator = (username, x, sess) -> false;
+        assertFalse(authenticate());
+    }
+
+    @Test
+    public void testSyncAuthSucceeded() throws Exception {
+        startServer();
+        authenticator = (username, x, sess) -> true;
+        assertTrue(authenticate());
+    }
+
+    @Test
+    public void testAsyncAuthFailed() throws Exception {
+        startServer();
+        authenticator = (username, x, sess) -> async(200, false);
+        assertFalse(authenticate());
+    }
+
+    @Test
+    public void testAsyncAuthSucceeded() throws Exception {
+        startServer();
+        authenticator = (username, x, sess) -> async(200, true);
+        assertTrue(authenticate());
+    }
+
+    @Test
+    public void testAsyncAuthTimeout() throws Exception {
+        startServer(500);
+        authenticator = (username, x, sess) -> asyncTimeout();
+        try {
+            authenticate();
+        } catch (JSchException e) {
+            assertTrue("Unexpected failure " + e.getMessage(), e.getMessage().startsWith("SSH_MSG_DISCONNECT"));
+        }
+    }
+
+    @Test
+    public void testAsyncAuthSucceededAfterTimeout() throws Exception {
+        startServer(500);
+        authenticator = (username, x, sess) -> async(1000, true);
+        try {
+            authenticate();
+        } catch (JSchException e) {
+            assertTrue("Unexpected failure " + e.getMessage(), e.getMessage().startsWith("SSH_MSG_DISCONNECT"));
+        }
+    }
+
+    private boolean asyncTimeout() {
+        throw new AsyncAuth();
+    }
+
+    private boolean async(int delay, boolean result) {
+        AsyncAuth auth = new AsyncAuth();
+        new Thread(() -> {
+            try {
+                Thread.sleep(delay);
+            } catch (InterruptedException ignore) {
+            } finally {
+                auth.setAuthed(result);
+            }
+        }).start();
+        throw auth;
```

Comment:
1. Let's make it a daemon thread
2. Add `// ignored` comment in the empty _catch_ clause